CONTRIBUTING.rst

```diff
@@ -0,0 +1,662 @@
+
+
+Pre-commit Hooks
+================
+
+Pre-commit hooks help speed up your local development cycle, either in the local virtualenv or Breeze,
+and place less burden on the CI infrastructure. Consider installing the pre-commit
+hooks as a necessary prerequisite.
+
+The pre-commit hooks only check the files you are currently working on and make
+them fast. Yet, these checks use exactly the same environment as the CI tests
+use. So, you can be sure your modifications will also work for CI if they pass
+pre-commit hooks.
+
+We have integrated the fantastic `pre-commit <https://pre-commit.com>`__ framework
+in our development workflow. To install and use it, you need Python 3.6 locally.
+
+It is the best to use pre-commit hooks when you have your local virtualenv for
+Airflow activated since then pre-commit hooks and other dependencies are
+automatically installed. You can also install the pre-commit hooks manually
+using ``pip install``.
+
+The pre-commit hooks require the Docker Engine to be configured as the static
+checks are executed in the Docker environment. You should build the images
+locally before installing pre-commit checks as described in `BREEZE.rst <BREEZE.rst>`__.
+In case you do not have your local images built, the
+pre-commit hooks fail and provide instructions on what needs to be done.
+
+Prerequisites for Pre-commit Hooks
+----------------------------------
+
+The pre-commit hooks use several external linters that need to be installed before pre-commit is run.
+
+Each of the checks installs its own environment, so you do not need to install those, but there are some
+checks that require locally installed binaries. On Linux, you typically install
+them with ``sudo apt install``, on macOS - with ``brew install``.
+
+The current list of prerequisites:
+
+-   ``xmllint``:
+    on Linux, install via ``sudo apt install xmllint``;
+    on macOS, install via ``brew install xmllint``
+
+Enabling Pre-commit Hooks
+-------------------------
+
+To turn on pre-commit checks for ``commit`` operations in git, enter:
+
+.. code-block:: bash
+
+    pre-commit install
+
+
+To install the checks also for ``pre-push`` operations, enter:
+
+.. code-block:: bash
+
+    pre-commit install -t pre-push
+
+
+For details on advanced usage of the install method, use:
+
+.. code-block:: bash
+
+   pre-commit install --help
+
+
+Using Docker Images for Pre-commit Hooks
+----------------------------------------
+
+Before running the pre-commit hooks, you must first build the Docker images as
+described in `BREEZE.rst <BREEZE.rst>`__.
+
+Sometimes your image is outdated and needs to be rebuilt because some
+dependencies have been changed. In such case the Docker-based pre-commit will
+inform you that you should rebuild the image.
+
+Supported Pre-commit Hooks
+--------------------------
+
+In Airflow, we have the following checks:
+
+.. code:: text
+
+  check-hooks-apply                   Checks which hooks are applicable to the repository.
```

Comment:
Why not a table? It is supported by docutils http://docutils.sourceforge.net/docs/ref/rst/directives.html#tables